lib/ace/ext/papeeria/tex_completer.coffee

```diff
@@ -126,19 +126,30 @@ define((require, exports, module) ->
     )
     editor.keyBinding.addKeyboardHandler(keyboardHandler)
 
-   class  ReferenceGetter
-    constructor: ->
+  processReferenceJson = (elem) =>
+    return {
+      name: elem.caption
+      value: elem.caption
+      meta: elem.type
+      meta_score: 10
+    }
+  # demo version
+  processCitationJson = (elem) =>
+    return {
+      name: elem.caption
+      value: elem.caption
+      meta: elem.type
+      meta_score: 10
+    }
+
+  class  PropertyGetter
```

Comment:
I am using this class for references and citations right know
You should use this class for \usepackage and end.
I thing that  ReferenceCache does not fit the context